core/ir/aarch64/codec.c

```diff
@@ -1963,10 +1963,38 @@ decode_opnd_imm16(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
     return decode_opnd_int(5, 16, false, 0, OPSZ_12b, 0, enc, opnd);
 }
 
+static bool
+encode_opnd_instr(int bit_pos, opnd_t opnd, byte *start_pc, instr_t *containing_instr,
+                  OUT uint *enc_out)
+{
+    if (!opnd_is_instr(opnd)) {
+        return false;
+    }
+    ptr_uint_t val =
+        ((ptr_uint_t)instr_get_note(opnd_get_instr(opnd)) -
+         (ptr_uint_t)instr_get_note(containing_instr) + (ptr_uint_t)start_pc) >>
+        opnd_get_shift(opnd);
+
+    uint bits = opnd_size_in_bits(opnd_get_size(opnd));
+    // We expect truncation; the caller splits up the instr's encoded address into
```

Comment:
I should probably clarify this comment: I don't mean to say that the caller of encode_opnd_instr does the splitting, but that it is done while creating the instr. Will push it with other requested changes (if any after this round of review).

Reply:
Done.